src/addagent/manage_agents.c

```diff
@@ -220,9 +240,80 @@ int add_agent(int json_output)
             _ip = NULL;
             c_ip.ip = NULL;
         } else if (!authd_running && (id_exist = IPExist(ip))) {
-            double antiquity = OS_AgentAntiquity_ID(id_exist);
+            bool replace_agent = false;
+            char error_message[OS_SIZE_128];
+
+            cJSON *j_agent_info = NULL;
+            j_agent_info = wdb_get_agent_info(atoi(id_exist), NULL);
+
+            if (!j_agent_info){
+                merror_exit("Failed to get agent info for agent '%s'", id_exist);
+            }
+
+            /* Check if the agent has been disconnected longer than the value required*/
+            if (env_disconnected_time) {
+                cJSON *j_connection_status = NULL;
+                cJSON *j_disconnected_time = NULL;
+                time_t agent_time_since_desconnection = 0;
+                char *status;
+
+                j_connection_status = cJSON_GetObjectItem(j_agent_info->child, "connection_status");
+                if(!j_connection_status){
+                    merror_exit("Failed to get connection status for agent '%s'.", id_exist);
```

Comment:
This exit will lead to an unoffensive memory leak. The application will shut down and free the memory, but any tool will detect this as a real memory leak, we should avoid it.

Reply:
There are other **exit** below. I suggest you to obtain all the possible fields right after **wdb_get_agent_info()**
And make one unique sanity check of the received info:
Like:

```
j_connection_status = cJSON_GetObjectItem(j_agent_info->child, "connection_status");
j_disconnected_time = cJSON_GetObjectItem(j_agent_info->child, "disconnected_time");
j_date_add = cJSON_GetObjectItem(j_agent_info->child, "date_add");
if(!j_connection_status || !j_disconnected_time  || !j_date_add ){
    cJSON_Delete(j_agent_info);
    merror_exit("Failed to parse agent-info for agent '%s'", id_exist);
}
```

It's true that we are extracting values that might be unnecessary, but this isn´t expensive and will simplify the code.

Reply:
Now all the checks are performed before.